wp-includes/class-wp-service-workers.php

```diff
@@ -63,6 +134,72 @@ public function init() {
 		do_action_ref_array( 'wp_default_service_workers', array( &$this ) );
 	}
 
+	/**
+	 * Configure offline page based on the saved offline page value.
+	 *
+	 * @return string Script.
+	 */
+	protected function configure_offline_page() {
```

Comment:
This should ideally tie-in to a precaching API (e.g. `register_precaching_for_routes`) instead of having its own separate logic.

Reply:
Updated the logic within dec04fb to use `register_precached_routes` for the precaching part.

Wondering if we should rename `register_precaching_for_routes` to `get_precaching_for_routes_script` or something similar, otherwise maybe it's too similar with `register_precached_routes`. Also, one is registering the route for caching and the other is generating and returning the script. Thoughts?

Reply:
Yes, good idea.

Reply:
Updated within a71a0cd.